buildpack/boreal/index.js

```diff
@@ -98,7 +98,8 @@ exports.processSourcePayload = async payload => {
     }
 
     const request = new ServerRequest(body, { headers, url });
-    await fn(request, payload.settings);
+    let settings = payload.hasOwnProperty("settings") ? payload.settings : {};
```

Comment:
You can use 
` let settings = payload.settings || {};`

Because hasOwnProperty  only checks if property is assigned to the object (even if is undefined)
```
let payload= {settings: undefined };
let settings2 = payload.hasOwnProperty("settings") ? payload.settings : {};
// settings2 is undefined 
let settings1= payload.settings || {};
// settings1 will be {}
```

Reply:
Thanks @kellymtrinh! Done: https://github.com/segmentio/functions-library/pull/33/commits/7a291f68dd49ff189f3bcc68add3ed5db393fd1b